pd/src/info/tendermint_proxy.rs

```diff
@@ -0,0 +1,359 @@
+        req: tonic::Request<AbciQueryRequest>,
+    ) -> Result<tonic::Response<AbciQueryResponse>, Status> {
+        // generic bounds on HttpClient::new are not well-constructed, so we have to
+        // render the URL as a String, then borrow it, then re-parse the borrowed &str
+        let client = HttpClient::new(self.tendermint_url.to_string().as_ref()).unwrap();
+
+        let path = Path::from_str(&req.get_ref().path).map_err(|_| tonic::Status::invalid_argument("invalid abci path"))?;
+        let data = &req.get_ref().data;
+        let height: Height = req
+            .get_ref()
+            .height
+            .try_into().map_err(|_| tonic::Status::invalid_argument("invalid height"))?;
+        let prove = req.get_ref().prove;
+        let res = client
+            .abci_query(Some(path), data.clone(), Some(height), prove)
+            .await.map_err(|e| tonic::Status::unavailable(format!(
+                    "error querying abci: {}",
+                    e
+                )))?;
+
+        match res.code {
+            tendermint_rpc::abci::Code::Ok => Ok(tonic::Response::new(AbciQueryResponse {
+                code: u32::from(res.code),
+                log: res.log.to_string(),
+                info: res.info,
+                index: res.index,
+                key: res.key,
+                value: res.value,
+                proof_ops: res
+                    .proof
+                    .map(|p| penumbra_proto::tendermint_proxy::ProofOps {
+                        ops: p
+                            .ops
+                            .into_iter()
+                            .map(|op| penumbra_proto::tendermint_proxy::ProofOp {
+                                r#type: op.field_type,
+                                key: op.key,
+                                data: op.data,
+                            })
+                            .collect(),
+                    }),
+                height: i64::try_from(res.height.value()).map_err(|_| {
+                    tonic::Status::internal(
+                        "height from tendermint overflowed i64, this should never happen",
+                    )
+                })?,
+                codespace: res.codespace,
+            })),
+            tendermint_rpc::abci::Code::Err(e) => Err(tonic::Status::unavailable(format!(
+                "error querying abci: {}",
+                e
+            ))),
+        }
+    }
+
+    async fn get_node_info(
+        &self,
+        req: tonic::Request<GetNodeInfoRequest>,
+    ) -> Result<tonic::Response<GetNodeInfoResponse>, Status> {
+        todo!()
+    }
+
+    async fn get_syncing(
+        &self,
+        req: tonic::Request<GetSyncingRequest>,
+    ) -> Result<tonic::Response<GetSyncingResponse>, Status> {
+        todo!()
+    }
+
+    async fn get_latest_block(
+        &self,
+        req: tonic::Request<GetLatestBlockRequest>,
+    ) -> Result<tonic::Response<GetLatestBlockResponse>, Status> {
+        todo!()
```

Comment:
We could replace some of these `todo!()`s with `tonic::Status::Unimplemented` and just only support the subset of methods that we actually use from `pcli`.

Reply:
So it turns out that we don't actually use _any_ of the methods defined in the trait from pcli.

Reply:
I think we do full block downloads in the view service?

Reply:
That's right, sorry. But we also call the "get status" call which isn't represented in the cosmos tendermint service definition